src/openrct2/ride/vehicle.h

```diff
@@ -221,41 +220,38 @@ typedef struct rct_vehicle_info {
-    VEHICLE_ENTRY_FLAG_A_NO_UPSTOP_WHEELS = 1 << 1,
-    VEHICLE_ENTRY_FLAG_A_NO_UPSTOP_BOBSLEIGH = 1 << 2,
-    VEHICLE_ENTRY_FLAG_A_MINI_GOLF = 1 << 3,
-    VEHICLE_ENTRY_FLAG_A_4 = 1 << 4,
-    VEHICLE_ENTRY_FLAG_A_5 = 1 << 5,
-    VEHICLE_ENTRY_FLAG_A_6 = 1 << 6,
-    VEHICLE_ENTRY_FLAG_A_7 = 1 << 7,
-    VEHICLE_ENTRY_FLAG_A_ALLOW_DOORS_DEPRECATED = 1 << 8,       // Not used any more - every vehicle will now work with doors
-    VEHICLE_ENTRY_FLAG_A_ENABLE_ADDITIONAL_COLOUR_2 = 1 << 9,
-    VEHICLE_ENTRY_FLAG_A_10 = 1 << 10,
-    VEHICLE_ENTRY_FLAG_A_11 = 1 << 11,
-    VEHICLE_ENTRY_FLAG_A_12 = 1 << 12,
-    VEHICLE_ENTRY_FLAG_A_13 = 1 << 13,
-    VEHICLE_ENTRY_FLAG_A_14 = 1 << 14,
-    VEHICLE_ENTRY_FLAG_A_15 = 1 << 15,
-};
-
-enum {
-    VEHICLE_ENTRY_FLAG_B_ENABLE_ADDITIONAL_COLOUR_1 = 1 << 0,
-    VEHICLE_ENTRY_FLAG_B_SWINGING = 1 << 1,
-    VEHICLE_ENTRY_FLAG_B_SPINNING = 1 << 2,
-    VEHICLE_ENTRY_FLAG_B_3 = 1 << 3,
-    VEHICLE_ENTRY_FLAG_B_RIDERS_SCREAM = 1 << 4,
-    VEHICLE_ENTRY_FLAG_B_5 = 1 << 5,
-    VEHICLE_ENTRY_FLAG_B_6 = 1 << 6,
-    VEHICLE_ENTRY_FLAG_B_7 = 1 << 7,
-    VEHICLE_ENTRY_FLAG_B_8 = 1 << 8,
-    VEHICLE_ENTRY_FLAG_B_9 = 1 << 9,
-    VEHICLE_ENTRY_FLAG_B_10 = 1 << 10,
-    VEHICLE_ENTRY_FLAG_B_11 = 1 << 11,
-    VEHICLE_ENTRY_FLAG_B_12 = 1 << 12,
-    VEHICLE_ENTRY_FLAG_B_13 = 1 << 13,
-    VEHICLE_ENTRY_FLAG_B_14 = 1 << 14,
-    VEHICLE_ENTRY_FLAG_B_15 = 1 << 15,
+    VEHICLE_ENTRY_FLAG_ENABLE_ADDITIONAL_COLOUR_1 = 1 << 0,
+    VEHICLE_ENTRY_FLAG_SWINGING = 1 << 1,
+    VEHICLE_ENTRY_FLAG_SPINNING = 1 << 2,
+    VEHICLE_ENTRY_FLAG_POWERED = 1 << 3,
+    VEHICLE_ENTRY_FLAG_RIDERS_SCREAM = 1 << 4,
+    VEHICLE_ENTRY_FLAG_5 = 1 << 5,
+    VEHICLE_ENTRY_FLAG_6 = 1 << 6,
+    VEHICLE_ENTRY_FLAG_7 = 1 << 7,
+    VEHICLE_ENTRY_FLAG_8 = 1 << 8,
+    VEHICLE_ENTRY_FLAG_9 = 1 << 9,
+    VEHICLE_ENTRY_FLAG_10 = 1 << 10,
+    VEHICLE_ENTRY_FLAG_11 = 1 << 11,
+    VEHICLE_ENTRY_FLAG_12 = 1 << 12,
+    VEHICLE_ENTRY_FLAG_13 = 1 << 13,
+    VEHICLE_ENTRY_FLAG_14 = 1 << 14,
+    VEHICLE_ENTRY_FLAG_15 = 1 << 15,
+    VEHICLE_ENTRY_FLAG_16 = 1 << 16,
+    VEHICLE_ENTRY_FLAG_NO_UPSTOP_WHEELS = 1 << 17,
+    VEHICLE_ENTRY_FLAG_NO_UPSTOP_BOBSLEIGH = 1 << 18,
+    VEHICLE_ENTRY_FLAG_MINI_GOLF = 1 << 19,
+    VEHICLE_ENTRY_FLAG_20 = 1 << 20,
+    VEHICLE_ENTRY_FLAG_21 = 1 << 21,
+    VEHICLE_ENTRY_FLAG_HAS_INVERTED_SPRITE_SET = 1 << 22,       // Set on vehicles that support running inverted for extended periods of time, i.e. the Flying, Lay-down and Multi-dimension RCs.
+    VEHICLE_ENTRY_FLAG_23 = 1 << 23,
+    VEHICLE_ENTRY_FLAG_ALLOW_DOORS_DEPRECATED = 1 << 24,        // Not used any more - every vehicle will now work with doors
+    VEHICLE_ENTRY_FLAG_ENABLE_ADDITIONAL_COLOUR_2 = 1 << 25,
+    VEHICLE_ENTRY_FLAG_26 = 1 << 26,
+    VEHICLE_ENTRY_FLAG_27 = 1 << 27,
+    VEHICLE_ENTRY_FLAG_28 = 1 << 28,
+    VEHICLE_ENTRY_FLAG_29 = 1 << 29,
+    VEHICLE_ENTRY_FLAG_30 = 1 << 30,
+    VEHICLE_ENTRY_FLAG_31 = 1u << 31,
```

Comment:
Yes, it is. It means: force this number to be an unsigned integer. This is necessary, because bit 31 of a _signed_ integer acts as the minus sign, not as part of the number. So this will ensure correct operation.